python/batch-predictor/docker/app.Dockerfile

```diff
@@ -13,19 +13,32 @@
 # limitations under the License.
 
 ARG BASE_IMAGE
-
 FROM ${BASE_IMAGE}
-# Get the model
 
-ARG MODEL_URL
 ARG GOOGLE_APPLICATION_CREDENTIALS
+RUN if [ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" ]; \
+    then gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS} > gcloud-auth.txt; \
+    else echo "GOOGLE_APPLICATION_CREDENTIALS is not set. Skipping gcloud auth command." > gcloud-auth.txt; \
+    fi
+
+# Download and install user model dependencies
+ARG MODEL_DEPENDENCIES_URL
+RUN gsutil cp ${MODEL_DEPENDENCIES_URL} conda.yaml
+RUN conda env create --name merlin-model --file conda.yaml
+
+# Copy and install batch predictor dependencies
+COPY --chown=${UID}:${GID} batch-predictor ${HOME}/merlin-spark-app
```

Comment:
1. Will this command reuse the cache if the Kaniko build context doesn't change (i.e., 2 image building jobs performed within the same release using the same user conda env file)?
Yes. It will reuse the cache if no change in the build context.

2. If the build context changes (eg: 0.36.0 -> 0.37.0), is it command sure to get the new code (0.37.0) instead of reusing cache?
I haven't tested it, but in theory, it should get the new code and invalidate the cache. Will get back to you once I tested it.

Reply:
Just tested it and the hypothesis is correct. The new code invalidates the cache.